cli/src/main/java/io/quarkiverse/tekton/cli/task/TaskExec.java

```diff
@@ -95,10 +96,11 @@ public void process(List<HasMetadata> resources) {
                 v1Task.getSpec().getParams().forEach(p -> parameters.put(p.getName(), p.getType()));
             }
 
-            List<Param> params = parameters.size() == 1
-                    ? Params.createSingle(parameters.keySet().iterator().next(), parameters.values().iterator().next(),
-                            taskArgs)
-                    : Params.create(taskArgs);
+            // Convert the arguments passed by the Cli command as List<String> into a Map<String,String> where the key is equal is the left part of key=val
```

Comment:
Same as above. If we move the logic inside `Params.create` we won't have to implement the splitting twice.

Reply:
Fixed: 06f79c5